jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -283,23 +283,47 @@ static void sendRedirect(Request request, Response response, Callback callback,
      */
     static void sendRedirect(Request request, Response response, Callback callback, int code, String location, boolean consumeAvailable)
     {
-        if (!HttpStatus.isRedirection(code))
+        sendRedirect(request, response, callback, code, location, consumeAvailable, null);
+    }
+
+    /**
+     * <p>Sends a {@code 302} HTTP redirect status code to the given location.</p>
+     *
+     * @param request the HTTP request
+     * @param response the HTTP response
+     * @param callback the callback to complete
+     * @param code the redirect HTTP status code, or 0 for a default
+     * @param location the redirect location as an absolute URI or encoded relative URI path.
+     * @param consumeAvailable whether to consumer the available request content
+     * @param content the content of the response, or null for a generated HTML message if {@link HttpConfiguration#isGenerateRedirectBody()} is {@code true}.
+     * @see #toRedirectURI(Request, String)
+     * @throws IllegalArgumentException if the status code is not a redirect, or the location is {@code null}
+     * @throws IllegalStateException if the response is already {@link #isCommitted() committed}
+     */
+    static void sendRedirect(Request request, Response response, Callback callback, int code, String location, boolean consumeAvailable, ByteBuffer content)
+    {
+        if (response.isCommitted())
         {
-            callback.failed(new IllegalArgumentException("Not a 3xx redirect code"));
+            callback.failed(new IllegalStateException("Committed"));
             return;
         }
 
-        if (location == null)
+        if (code <= 0)
+            code = HttpMethod.GET.is(request.getMethod()) || request.getConnectionMetaData().getHttpVersion().getVersion() < HttpVersion.HTTP_1_1.getVersion()
+            ? HttpStatus.MOVED_TEMPORARILY_302
+            : HttpStatus.SEE_OTHER_303;
+        if (!HttpStatus.isRedirection(code))
         {
-            callback.failed(new IllegalArgumentException("No location"));
+            callback.failed(new IllegalArgumentException("Not a 3xx redirect code"));
```

Comment:
The change to `isRedirection(int)` to not consider 304 a redirect means that this IAE can happen on a 304 now?

Of course, that would mean someone using sendRedirect with a 304 status code.  Which is just a strange thing to do.

Reply:
Considering also the check for `location` below, seems that there really is a class of "redirect to location" codes that is worth a method in `HttpStatus`.
And yes, an application calling this method with 304 should get a ISE.